### ISSUE TITLE
TypeError when prefer-for-of is enabled

### ISSUE THREAD

**Issue by buu700** [palantir/tslint#1753]
### Bug Report

- __TSLint version__: `4.0.0`
- __TypeScript version__: `2.0.10`
- __Running TSLint via__: CLI

#### TypeScript code being linted

Not yet entirely sure what code specifically is triggering this, but running `tslint --project shared/js/tsconfig.json -c tslint.json` in [cyph/cyph](/cyph/cyph) with the configuration below will reproduce it.

with `tslint.json` configuration:

```
{
	"rules": {
		"prefer-for-of": true
	}
}
```

#### Actual behavior

```
/usr/local/lib/node_modules/tslint/lib/rules/preferForOfRule.js:102
            if (arrayAccessNode.kind !== ts.SyntaxKind.PropertyAccessExpression && forLoop.initializer !== undefined) {
                               ^

TypeError: Cannot read property 'kind' of undefined
    at PreferForOfWalker.locateArrayNodeInForLoop (/usr/local/lib/node_modules/tslint/lib/rules/preferForOfRule.js:102:32)
    at PreferForOfWalker.visitForStatement (/usr/local/lib/node_modules/tslint/lib/rules/preferForOfRule.js:57:36)
    at PreferForOfWalker.SyntaxWalker.visitNode (/usr/local/lib/node_modules/tslint/lib/language/walker/syntaxWalker.js:343:22)
    at /usr/local/lib/node_modules/tslint/lib/language/walker/syntaxWalker.js:505:63
    at visitEachNode (/usr/local/lib/node_modules/typescript/lib/typescript.js:8343:30)
    at Object.forEachChild (/usr/local/lib/node_modules/typescript/lib/typescript.js:8501:24)
    at PreferForOfWalker.SyntaxWalker.walkChildren (/usr/local/lib/node_modules/tslint/lib/language/walker/syntaxWalker.js:505:12)
    at PreferForOfWalker.SyntaxWalker.visitBlock (/usr/local/lib/node_modules/tslint/lib/language/walker/syntaxWalker.js:47:14)
    at PreferForOfWalker.SyntaxWalker.visitNode (/usr/local/lib/node_modules/tslint/lib/language/walker/syntaxWalker.js:283:22)
    at /usr/local/lib/node_modules/tslint/lib/language/walker/syntaxWalker.js:505:63
```

#### Expected behavior

Either no output or a linting error.

**Comment by adidahiya** [comment]
are you running this on JS files or TS files? would be super helpful if you could isolate the failure down to one file.

**Comment by buu700** [comment]
TS (every other rule works as expected aside from this one, so it isn't a general problem with our code or project structure). Just dug a bit more into it, and it looks like these two files trigger the problem:

* [util.ts](/cyph/cyph/blob/master/shared/js/cyph/util.ts)

* [files.ts](/cyph/cyph/blob/master/shared/js/cyph/files/files.ts)

**Comment by adidahiya** [comment]
hm, interesting, this is probably a bug. we have no [tests](https://github.com/palantir/tslint/blob/ac05171f547a1b0711df7ff7301fd9bd5f5a7265/test/rules/prefer-for-of/test.ts.lint) for a loop like [this in utils.ts](https://github.com/cyph/cyph/blob/ce4878bb48db4c8e3fe33bf0cb7b0a86d048e38d/shared/js/cyph/util.ts#L65) where the index isn't referenced. but not sure what's going wrong in files.ts.

**Comment by nchen63** [comment]
crashes on `for (var i = 0; true; i++)`

**Comment by buu700** [comment]
re: files.ts, after looking at it again, my guess is that [this](https://github.com/cyph/cyph/blob/master/shared/js/cyph/files/files.ts#L102) is the problem.

**Comment by nchen63** [comment]
@buu700 fixed in linked PR

**Comment by raptoria** [comment]
has this fix been merged? i'm using version 2.16.0 and tested 2.17.0 and it still exists

**Comment by buu700** [comment]
It looks like that was made in tslint 4.x, not 2.x.
